src/table/components/HeadCellMenu.tsx

```diff
@@ -0,0 +1,99 @@
+import Grow from '@mui/material/Grow';
+import Paper from '@mui/material/Paper';
+import Popper from '@mui/material/Popper';
+import MenuList from '@mui/material/MenuList';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';
+import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward';
+import MoreHoriz from '@mui/icons-material/MoreHoriz';
+import { StyledMenuIconButton, StyledCellMenu } from '../styles';
+import { GeneratedStyling } from '../types';
+
+const ListItems = ({ children, itemTitle }: { children: any; itemTitle: string }) => {
+  return (
+    <ListItem disablePadding>
+      <ListItemButton>
+        <ListItemIcon>{children}</ListItemIcon>
+        <ListItemText primary={itemTitle} />
+      </ListItemButton>
+    </ListItem>
+  );
+};
+
+export default function HeadCellMenu({ headerStyle }: { headerStyle: GeneratedStyling }) {
+  const [open, setOpen] = useState(false);
+  const anchorRef = useRef<HTMLButtonElement>(null);
+  // It returns focus to the button when we transitioned from !open -> open
+  const prevOpen = useRef(open);
+
+  const handleClickOpen = (event: any) => {
+    event.stopPropagation();
+    setOpen((value) => !value);
+  };
+
+  const handleClickClose = (event: Event | React.SyntheticEvent) => {
+    !(anchorRef.current && anchorRef.current.contains(event.target as HTMLElement)) ? setOpen(false) : undefined;
+  };
+
+  useEffect(() => {
+    if (prevOpen.current === true && open === false) {
+      anchorRef.current!.focus();
+    }
+    prevOpen.current = open;
+  }, [open]);
+
+  return (
+    <StyledCellMenu headerStyle={headerStyle}>
+      <StyledMenuIconButton
+        ref={anchorRef}
+        id="head-cell-button"
+        aria-controls={open ? 'head-cell-menu' : undefined}
+        aria-expanded={open ? 'true' : undefined}
+        aria-haspopup="true"
+        onClick={handleClickOpen}
+      >
+        <MoreHoriz />
+      </StyledMenuIconButton>
+      <Popper
+        open={open}
+        anchorEl={anchorRef.current}
+        role={undefined}
+        placement="bottom-start"
+        transition
+        disablePortal
+      >
+        {({ TransitionProps, placement }) => (
+          <Grow
+            {...TransitionProps}
+            style={{
+              transformOrigin: placement === 'bottom-start' ? 'left top' : 'left bottom',
+            }}
+          >
+            <Paper sx={{ boxShadow: 15 }}>
+              <ClickAwayListener onClickAway={handleClickClose}>
+                <MenuList autoFocusItem={open} className="head-cell-menu" aria-labelledby="head-cell-button">
+                  <nav aria-label="Sorting orders">
+                    <List>
+                      <ListItems itemTitle="Sort ascending">
```

Comment:
will need to add proper strings for this eventually

Reply:
Do you mean the itemTitle text? It is what design team has chosen.

Reply:
yes but we need to add it to translation, so we call translator.get('somStringName') so it works for all languages

Reply:
Should it be added in sense-client or there is a place in sn-table repo for it ?